### PR TITLE
Add support for SSO to parent and child bot projects for manual testing

### PR DESCRIPTION
**Pull request by swagatmishra2007**
- Add support for SSO to parent and child bot projects for manual testing
- Fixed  a minor bug in oauthprompt

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs [hunk 1]
```diff
@@ -458,19 +458,6 @@ private async Task<PromptRecognizerResult<TokenResponse>> RecognizeTokenAsync(IT
                             FailureDetail = "The bot received an InvokeActivity that is missing a TokenExchangeInvokeRequest value. This is required to be sent with the InvokeActivity.",
                         }).ConfigureAwait(false);
                 }
-                else if (tokenExchangeRequest.ConnectionName != _settings.ConnectionName)
-                {
-                    await this.SendInvokeResponseAsync(
-                        turnContext,
-                        cancellationToken,
-                        HttpStatusCode.BadRequest,
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs]: Please confirm with @Jeffders that we can remove this.

File: tests/Skills/Parent/ParentBot.cs [hunk 2]
```diff
@@ -46,5 +120,75 @@ protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivi
...
+                _fromBotId,
+                _toBotId,
+                new Uri("http://localhost:2303/api/messages"),
+                new Uri("http://tempuri.org/whatever"),
+                incomingActivity.Conversation.Id,
+                activity,
+                cancellationToken);
+
+            // Check response status: true if success, false if failure
+            var success = response.Status >= 200 && response.Status <= 299;
```
Comment (carlosscastro) [inline tests/Skills/Parent/ParentBot.cs]: IsSuccessstatusCode

**Comment by carlosscastro** [comment]
Approved but please check in person with @Jeffders  as of why he added that codepath. The fact that there is a test indicates that it is on purpose, so there might be some scenario that we are missing.

File: libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs [hunk 3]
```diff
@@ -458,19 +458,6 @@ private async Task<PromptRecognizerResult<TokenResponse>> RecognizeTokenAsync(IT
...
-                        turnContext,
-                        cancellationToken,
-                        HttpStatusCode.BadRequest,
-                        new TokenExchangeInvokeResponse()
-                        {
-                            Id = tokenExchangeRequest.Id,
-                            ConnectionName = _settings.ConnectionName,
-                            FailureDetail = "The bot received an InvokeActivity with a TokenExchangeInvokeRequest containing a ConnectionName that does not match the ConnectionName expected by the bot's active OAuthPrompt. Ensure these names match when sending the InvokeActivityInvalid ConnectionName in the TokenExchangeInvokeRequest",
-                        }).ConfigureAwait(false);
-                }
```
Comment (swagatmishra2007) [inline libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs]: revert

**Pull request merged**